jsinterop-ts-defs-processor/src/main/java/com/vertispan/tsdefs/model/TsUnionType.java

```diff
@@ -16,26 +16,40 @@
 package com.vertispan.tsdefs.model;
 
 import java.util.Arrays;
-import java.util.List;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 public class TsUnionType extends TsType {
 
-  private List<TsType> tsTypes;
+  private final Set<TsType> tsTypes = new LinkedHashSet<>();
 
-  public TsUnionType(List<TsType> tsTypes) {
+  public TsUnionType(Set<TsType> tsTypes) {
     super("", "");
-    this.tsTypes = tsTypes;
+
+    tsTypes.forEach(
+        tsType -> {
+          if (tsType instanceof TsUnionType) {
+            this.tsTypes.addAll(((TsUnionType) tsType).tsTypes);
+          } else {
+            this.tsTypes.add(tsType);
+          }
+        });
```

Comment:
could also be this, remove the initialized value on the field

```suggestion
    this.tsTypes = tsTypes.stream()
        .flatMap(type -> {
          if (type instanceof TsUnionType) {
            return ((TsUnionType) type).tsTypes.stream();
          }
          return Stream.of(tsType);
        })
        .collect(Collectors.toCollection(LinkedHashSet::new));
```
or something in between here and there